extensions/godslayerakp/ws.js

```diff
@@ -0,0 +1,283 @@
+            blockType: BlockType.EVENT,
+            isEdgeActivated: false,
+            text: "when connection connects",
+          },
+          {
+            opcode: "isConnected",
+            blockType: BlockType.BOOLEAN,
+            text: "is connection connected?",
+          },
+          "---",
+          {
+            opcode: "onMessage",
+            blockType: BlockType.EVENT,
+            isEdgeActivated: false,
+            text: "when message recieved",
+          },
+          {
+            opcode: "messageReceived",
+            blockType: BlockType.BOOLEAN,
+            text: "recieved message?",
+          },
+          {
+            opcode: "messageData",
+            blockType: BlockType.REPORTER,
+            disableMonitor: true,
+            text: "recieved message data",
+          },
+          {
+            opcode: "sendMessage",
+            blockType: BlockType.COMMAND,
+            arguments: {
+              PAYLOAD: {
+                type: ArgumentType.STRING,
+                defaultValue: "hello!",
+              },
+            },
+            text: "send message [PAYLOAD]",
+          },
+          "---",
+          {
+            opcode: "onClose",
+            blockType: BlockType.EVENT,
+            isEdgeActivated: false,
+            text: "when connection closes",
+          },
+          {
+            opcode: "isClosed",
+            blockType: BlockType.BOOLEAN,
+            text: "is connection closed?",
+          },
+          {
+            opcode: "closeCode",
+            blockType: BlockType.REPORTER,
+            disableMonitor: true,
+            text: "closing code",
+          },
+          {
+            opcode: "closeMessage",
+            blockType: BlockType.REPORTER,
+            disableMonitor: true,
+            text: "closing message",
+          },
+          {
+            opcode: "closeWithoutReason",
+            blockType: BlockType.COMMAND,
+            text: "close connection without reason",
+          },
+          {
+            opcode: "closeWithCode",
+            blockType: BlockType.COMMAND,
+            arguments: {
+              CODE: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "1000",
+              },
+            },
+            text: "close connection with code [CODE]",
+          },
+          {
+            opcode: "closeWithReason",
+            blockType: BlockType.COMMAND,
+            arguments: {
+              CODE: {
+                type: ArgumentType.NUMBER,
+                defaultValue: "1000",
+              },
+              REASON: {
+                type: ArgumentType.STRING,
+                defaultValue: "connection fulfilled",
+              },
+            },
+            text: "close connection with reason [REASON] and code [CODE]",
+          },
+        ],
+      };
+    }
+    async newInstance(args, utils) {
+      let url = Cast.toString(args.URL);
+      if (!url.startsWith("ws")) {
```

Comment:
This will fail on inputs like `wstestexample.com`

Reply:
> This will fail on inputs like `wstestexample.com`

...
yeah i didnt think about that